### ISSUE TITLE
Different declarations with the same signatures were detected after upgrading from 1.0.1 to 1.1.0

### ISSUE THREAD

**Issue by PhilippNowak96** [InsertKoinIO/koin-compiler-plugin#84]
For some internal reasons I have two separate Ktor HTTP clients and my module looks like this:

```kotlin
val networkModule = module {
    single<HttpClient>(named("httpClient")) {
        createHttpClient(
            json = get(),
            userStore = get(),
            tokenApi = get(),
        )
    }
    single<HttpClient>(named("tokenClient")) {
        createTokenHttpClient(json = get())
    }
}
```

In 1.0.1, everything worked fine. After upgrading to 1.1.0, I get the following error:

```
error: java.lang.AssertionError: Different declarations with the same signatures were detected, but no diagnostics will be reported because none of those declarations have any source location info associated with them. This could happen because the declarations came from an external module, or were generated by a compiler plugin. The following declarations have conflicting signatures:
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:io.ktor.client.HttpClient, module_com$xxx$core$di$networkModule:kotlin.Unit, providerOnly:kotlin.Unit, qualifier_httpClient:kotlin.Unit) returnType:kotlin.Unit
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:io.ktor.client.HttpClient, module_com$xxx$core$di$networkModule:kotlin.Unit, providerOnly:kotlin.Unit, qualifier_tokenClient:kotlin.Unit) returnType:kotlin.Unit
```

Not sure if it is a bug or on my end.

**Comment by arnaudgiuliani** [comment]
can you reproduce on 1.2.1 ?

**Comment by OscarD8** [comment]
Just to add, I can reproduce this on 1.1.0 and it looks to be resolved on 1.2.1.

I had the same error with a different qualifier setup (Qualifier annotations instead of named()).

**Setup**:

- Kotlin: 2.4.10
- AGP: 9.4.0
- Koin Core / Annotations: 4.2.2
- Koin Compiler Plugin: 1.1.0 (fails) vs 1.2.1 (passes)
- Targets: LinuxX64, Android, JVM



Qualifier annotations:

```kotlin
@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FUNCTION, AnnotationTarget.TYPE)
@Retention(AnnotationRetention.BINARY)
@Qualifier
annotation class IoContext

@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FUNCTION, AnnotationTarget.TYPE)
@Retention(AnnotationRetention.BINARY)
@Qualifier
annotation class MainContext

@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FUNCTION, AnnotationTarget.TYPE)
@Retention(AnnotationRetention.BINARY)
@Qualifier
annotation class DefaultContext
```

DSL module registering multiple instances of the same type (CoroutineContext), differentiated by qualifier:

```kotlin
fun TestScope.bindAsKoinModule(): Module = module {
    single<CoroutineContext> { dispatcher }
    single<CoroutineContext>(qualifier<IoContext>()) { dispatcher }
    single<CoroutineContext>(qualifier<MainContext>()) { dispatcher }
    single<CoroutineContext>(qualifier<DefaultContext>()) { dispatcher }
}
```

**On 1.1.0:**

Kotlin/Native (compileKotlinLinuxX64) fails:

```
e: java.lang.AssertionError: Different declarations with the same signatures were detected, but no diagnostics will be reported because none of those declarations have any source location info associated with them. This could happen because the declarations came from an external module, or were generated by a compiler plugin. The following declarations have conflicting signatures:
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:kotlin.coroutines.CoroutineContext, providerOnly:kotlin.Unit) returnType:kotlin.Unit
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:kotlin.coroutines.CoroutineContext, providerOnly:kotlin.Unit) returnType:kotlin.Unit
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:kotlin.coroutines.CoroutineContext, providerOnly:kotlin.Unit) returnType:kotlin.Unit
 - FUN name:dsl_single visibility:public modality:FINAL <> (contributed:kotlin.coroutines.CoroutineContext, providerOnly:kotlin.Unit) returnType:kotlin.Unit

        at org.jetbrains.kotlin.backend.common.linkage.issues.SignatureClashDetector.reportSignatureClashTo(SignatureClashDetector.kt:71)
        at org.jetbrains.kotlin.backend.common.diagnostics.IdSignatureClashDetector.reportSignatureConflict(IdSignatureClashDetector.kt:20)
```

Android target (connectedAndroidDeviceTest) warns about multiple definitions, but compiles successfully:

```
WARNING: D8: Ignoring an implementation of the method `void org.koin.plugin.hints...dsl_single(kotlin.coroutines.CoroutineContext, kotlin.Unit)` because it has multiple definitions
```

JVM (jvmTest) compiles successfully with no warning.

**On 1.2.1:**

Compiles successfully on all targets and tests run as expected (same as version 1.0.1).

```
w: [Koin] Intercepting single<CoroutineContext> { ... } (provider-only)
w: [Koin] Intercepting single<CoroutineScope> { ... } (provider-only)
```

Sharing just in case its useful. Happy to provide any more info.
